cmd/mirror.go

```diff
@@ -221,80 +118,193 @@ func newMirrorSetCmd() *cobra.Command {
 	return cmd
 }
 
+// the `mirror introduce` sub command
+func newMirrorIntroduceCmd() *cobra.Command {
+	name := ""
+	privPath := ""
+
+	cmd := &cobra.Command{
+		Use:   "introduce <id>",
+		Short: "introduce a new owner",
+		Long:  "introduce a new owner to current mirror",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) < 1 {
+				return cmd.Help()
+			}
+
+			id := args[0]
+			if name == "" {
+				fmt.Printf("The --name hasn't specified value, default to %s\n", id)
+				name = id
+			}
+
+			// the privPath can point to a public key becase the Public method of KeyInfo works on both priv and pub key
+			privKey, err := loadPrivKey(privPath)
+			if err != nil {
+				return err
+			}
+			pubKey, err := privKey.Public()
+			if err != nil {
+				return err
+			}
+
+			env := environment.GlobalEnv()
+			return env.V1Repository().Mirror().Introduce(id, name, pubKey)
+		},
+	}
+
+	cmd.Flags().StringVarP(&name, "name", "n", "", "Specify the name of the owner, default: id of the owner")
+	cmd.Flags().StringVarP(&privPath, "key", "k", "", "Specify the private key path of the owner")
+
+	return cmd
+}
+
 // the `mirror modify` sub command
 func newMirrorModifyCmd() *cobra.Command {
 	var privPath string
-	endpoint := ""
 	desc := ""
 	standalone := false
 	hidden := false
 	yanked := false
 
 	cmd := &cobra.Command{
-		Use:  "modify <component>[:version] [flags]",
-		Long: "modify component attributes (hidden, standalone, yanked)",
+		Use:   "modify <component>[:version] [flags]",
+		Short: "Modify published component",
+		Long:  "Modify component attributes (hidden, standalone, yanked)",
 		RunE: func(cmd *cobra.Command, args []string) error {
 			if len(args) != 1 {
 				return cmd.Help()
 			}
-			env := environment.GlobalEnv()
-			if privPath == "" {
-				privPath = env.Profile().Path(localdata.KeyInfoParentDir, "private.json")
-			}
 
-			// Get the private key
-			f, err := os.Open(privPath)
-			if err != nil {
-				return err
-			}
-			defer f.Close()
+			component := args[0]
 
-			ki := v1manifest.KeyInfo{}
-			if err := json.NewDecoder(f).Decode(&ki); err != nil {
-				return err
-			}
+			env := environment.GlobalEnv()
 
-			comp, ver := environment.ParseCompVersion(args[0])
+			comp, ver := environment.ParseCompVersion(component)
 			m, err := env.V1Repository().FetchComponentManifest(comp, true)
 			if err != nil {
 				return err
 			}
 
-			if endpoint == "" {
-				endpoint = environment.Mirror()
+			v1manifest.RenewManifest(m, time.Now())
+			m.Version++
```

Comment:
Why not bump version in `RenewManifest()`?

Reply:
good idea